### PR TITLE
[FIX] point_of_sale: allow regular pos users cash in / out

### PR DESCRIPTION
**Pull request by chienandalu**
Description of the issue/feature this PR addresses:

Cash control is a basic operation in a point of sale and it shouldn't require permissions as big as Accounting/Billing to be performed

Current behavior before PR:

- A Point of Sale user with no Accounting/Billing permission can't do cash operations.

Desired behavior after PR is merged:

- Recover the paradigm present in versions <=v15: a pos user can record a cash in / out operation without the need of elevated permissions.

TT51987 cc @Tecnativa

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/models/pos_session.py [hunk 1]
```diff
@@ -1559,7 +1559,7 @@ def try_cash_in_out(self, _type, amount, reason, extras):
         if not sessions:
             raise UserError(_("There is no cash payment method for this PoS Session"))
 
-        self.env['account.bank.statement.line'].create([
+        self.env['account.bank.statement.line'].sudo().create([
```
Comment (vlst-odoo) [inline addons/point_of_sale/models/pos_session.py:1562]: can you check that the user at least has the `pos_user` rights before doing `sudo` ?

Comment (chienandalu) [reply addons/point_of_sale/models/pos_session.py:1562]: The `pos.session` model has it implicit as its only allowed group is already `point_of_sale.group_pos_user`. So this method will fail for anyone else. Should we do that explicit check?

**Review by vlst-odoo — APPROVED** [review]
robodoo r+

**Comment by vlst-odoo** [comment]
robodoo rebase-ff

**Pull request closed without merging**